src/app/shared/model/campaign.model.ts

```diff
@@ -30,31 +36,32 @@ export class CampaignModel extends BaseModel {
   related(): {} {
     let sponsor = Observable.of(null);
     if (this.doc && this.doc.has('prx:sponsor')) {
-      sponsor = this.doc.follow('prx:sponsor').map(sDoc => new SponsorModel(sDoc));
+      sponsor = this.doc.follow('prx:sponsor').map(sDoc => new SponsorModel(null, sDoc));
     }
     return { sponsor: sponsor };
   }
 
   decode(): void {
     this.id = this.doc['id'];
+    this.originalCopy = this.doc['originalCopy'];
+    this.editedCopy = this.doc['editedCopy'] || this.doc['mustSay'];
+    this.mustSay = this.doc['mustSay'];
+    this.notes = this.doc['notes'];
+    this.zone = this.doc['zone'];
     this.startDate = new Date(this.doc['startDate']);
     this.endDate = new Date(this.doc['endDate']);
-    this.copy = this.doc['copy'];
-    this.zone = this.doc['zone'];
-    this.dueDate = this.doc['dueDate'] ? new Date(this.doc['dueDate']) : new Date(); // TODO add due date to campaigns
-    this.approved = this.doc['approved']; // TODO add approved to campaigns
+    this.dueDate = toLocal(new Date(this.doc['dueDate']));
```

Comment:
I'm not sure I've followed all your date woes very closely. Timezones are a headache. The date is stored in a timezone independent way right? So when the browser does toDateString or whatever, it's shown in the local timezone. That much is built in. My headaches with utc dates on metrics was that I needed utc date string and to do date math in utc to make sure I'm starting and landing in the correct month/day. I'm gonna go ahead and trust that you've worked all this out for what's going on in here because I know you've had to wrestle with it for a bit.